### ISSUE TITLE
offsetHeight is 0 during componentDidMount with <style jsx>

### ISSUE THREAD

**Issue by chrillo** [zeit/styled-jsx#140]
```
class Component extends React.Component{
  componentDidMount(){

      console.log(this.refs.test.offsetHeight) // 0
      setTimeout(()=>{
        console.log(this.refs.test.offsetHeight) // 42
      },1)
  
  }
  render(){
      return (<div ref="test" className="test">
          <style jsx>{` // without jsx = regular style sheet it works

              .test{
                height: 42px;
              }
          
          `}</style>
      </div>)
  }
}
```

removing the jsx = using a normal style tag or external css file works. this seems to be some sort of race condition in how the jsx transpiled style tags get appended to the DOM. for 99% of sites this does not matter but for JS based layout computation this is a real dealbreaker.  not sure if thats even fixable but it would be nice to get some insight into why that is.

**Comment by giuseppeg** [comment]
@chrillo it should be related to #136

**Comment by chrillo** [comment]
@giuseppeg yes almost 100% sure it is, feel free to close this as a duplicate. I'll follow the other issue. thanks!!

**Comment by giuseppeg** [comment]
Fixed in #139
